bin/publish-modules

```diff
@@ -0,0 +1,93 @@
+#!/usr/bin/env node
+
+var exec = require( 'child-process-promise' ).exec;
+var semver = require( 'semver' );
+
+console.log( 'Modules requiring publish:' );
+
+function getAllModules() {
+	var command = 'find client -type f -print | grep \'package.json\'';
+	return exec( command ).then( function( result ) {
+		return result.stdout.split( /\s+/ );
+	} );
+}
+
+function getPublishableModules( allModules ) {
+	var packageJson;
+	return allModules.filter( function( modulePath ) {
+		if ( !modulePath ) {
+			return;
+		}
+		packageJson = require( modulePath );
+		if ( !packageJson.private ) {
+			return true;
+		}
+	} );
+}
+
+function publishModule( modulePath ) {
+	return exec( 'make -C ' + modulePath.slice( 0, -12 ) + ' publish' )
```

Comment:
Unfortunately prepublish runs when you npm install a package from a local path:  https://github.com/npm/npm/issues/3059  That makes it difficult to use.  To have that work all sub packages would have to have their dependencies installed.